x-pack/platform/packages/shared/kbn-ai-tools/src/tools/describe_dataset/sort_and_truncate_analyzed_fields.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { partition, shuffle } from 'lodash';
+import { truncateList } from '@kbn/inference-common';
+import type { DocumentAnalysis, TruncatedDocumentAnalysis } from './document_analysis';
+
+export function sortAndTruncateAnalyzedFields(
+  analysis: DocumentAnalysis,
+  options: { dropEmpty?: boolean } = {}
+): TruncatedDocumentAnalysis {
+  const { dropEmpty = false } = options;
+  const { fields, ...meta } = analysis;
+  const [nonEmptyFields, emptyFields] = partition(analysis.fields, (field) => !field.empty);
+
+  const sortedFields = [...shuffle(nonEmptyFields), ...shuffle(emptyFields)];
```

Comment:
Can you add a comment to note why we are calling shuffle() here?

Reply:
yes! it's to get a random-ish subset of fields, so you don't end up with a-f and drop everything on the floor. (i'll add this as a comment)